src/rebar3_hex_build.erl

```diff
@@ -1,3 +1,88 @@
+% @doc `rebar3 hex build' 
+%
+%% Builds a new local version of your package.
+%%
+%% By default this provider will build both a package tarball and docs tarball. 
+%%  
+%% The package and docs .tar files are created in the current directory, but is not pushed to the repository. An app 
+%% named foo at version 1.2.3 will be built as foo-1.2.3.tar. Likewise the docs .tar would be built as 
+%% foo-1.2.4-docs.tar. 
+%%
+%% ```shell
+%% $ rebar3 hex build 
+%% '''
+%% 
+%% You may also build only a package or docs tarball utilizing the same available command line options. 
+%%
+%% ``` shell
+%% $ rebar3 hex build package 
+%% '''
+%%
+%% ```
+%% $ rebar3 hex build docs 
+%% '''
+%% 
+%% <h2>Configuration</h2> 
+%% Packages are configured via `src/<myapp>.app.src'  attributes.
+%% 
+%% == Required configuration == 
+%% 
+%% <ul>
+%%  <li> `application' - application name. This is required per Erlang/OTP thus it should always be present anyway. 
+%   <li> `vsn' -  must be a valid [semantic version](http://semver.org/) identifier.
+%%  <li>`licenses' - A list of licenses the project is licensed under. This attribute is required. A valid
+%%  [spdx](https://spdx.org/licenses/) is expected.</li>
+%% </ul>
+
+%% 
+%% == Optional configuration == 
+%% In addition, the following meta attributes are supported and highly recommended : 
+%%
+%% <ul>
+%%  <li> `description' - a brief description about your application.
+%%  <li>`pkg_name' - The name of the package in case you want to publish the package with a different name than the
+%%  application name.</li>
+%%  <li>`links' - A map where the key is a link name and the value is the link URL. Optional but highly recommended.
+%%  <li> `files' - A list of files and directories to include in the package. Defaults to standard project directories, 
+%%        so you usually don't need to set this property.</li>
+%%  <li> `include_files' - A list of paths containing files you wish to include in a release. </li>
+%%  <li> `exclude_files' - A list of paths containing files you wish to exclude in a release. </li>
+%%  <li> `exclude_patterns' - A list of regular expressions used to exclude files that may have been accumulated via
+%%  `files' and `include_files' and standard project paths. 
+%%  <li> `build_tools' - List of build tools that can build the package. It's very rare that you need to set this. </li>
+%% </ul>
+%%
+%% Below is an example : 
+%%
+%% ```erlang
+%% {application, myapp,
+%%  [{description, "An Erlang/OTP application"},
+%%   {vsn, "0.1.0"},
+%%   {modules, []},
+%%   {registered, []},
+%%   {applications, [kernel,
+%%                   stdlib,
+%%                   ]},
+%%   {licenses, ["Apache-2.0"]},
+%%   {links, [{"GitHub", "https://github.com/my_name/myapp"}]}]}.
+%% ```
+%%
+%% <h2> Command line options </h2>
+%%
+%% <ul>
+%%  <li>`--repo' - Specify the repository to use in the task. This option is required when 
```

Comment:
I don't like that we support this on this provider, but we currently support being able to configure a doc provider on a per repo basis.